tutorial6_hierarchical/demo/main_advanced.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * CONSTANTS AND GLOBALS
+ * */
+const width = window.innerWidth * 0.9,
+  height = window.innerHeight * 0.7,
+  margin = { top: 20, bottom: 50, left: 60, right: 40 };
+
+let svg;
+
+/**
+ * APPLICATION STATE
+ * */
+let state = {
+  root: null,
+};
+
+/**
+ * LOAD DATA
+ * */
+d3.csv("../../data/netflix_titles_with_genre.csv", d3.autotype).then(data => {
+  state.data = data;
+  init();
+});
+
+/**
+ * INITIALIZING FUNCTION
+ * this will be run *one time* when the data finishes loading in
+ * */
+function init() {
+  const container = d3.select("#d3-container").style("position", "relative");
+
+  tooltip = container
+    .append("div")
+    .attr("width", 100)
+    .attr("height", 100)
+    .style("position", "absolute")
+    .style("background-color", "white");
+
+  svg = container
+    .append("svg")
+    .attr("width", width)
+    .attr("height", height);
+
+  const rolledUp = d3.rollups(
+    state.data,
+    v => ({ count: v.length, movies: v }), // reduce function,
+    d => d.genre,
+    d => d.type,
+    d => d.rating
+  );
+
+  console.log("rolledUp", rolledUp);
+  // groups the data by genre, type and rating
+  // make hierarchy
+  const root = d3
+    .hierarchy([null, rolledUp], ([key, values]) => values) // children accessor, tell it to grab the second element
+    .sum(([key, values]) => values.count) // sets the 'value' of each level
+    .sort((a, b) => b.value - a.value);
```

Comment:
got a working version but it isn't super elegant... Do you think we should just delete this `advanced` file?

Reply:
probably... I doubt I'll have time to discuss it, and they can find more advanced examples on their own if they are looking